systemtests/generate_testdata_on_network.py

```diff
@@ -66,204 +68,300 @@ def do_setup(args: Any):
-            sender=accounts.get_user(shard=0, index=0),
-            contract=accounts.get_contract_address("adder", 0, 0),
-        ))
-
-        # TODO: claim developer rewards with parent-child contracts
-        # TODO: claim developer rewards on directly owned contracts, cross shard (use change owner address).
-
-    print("Intra-shard, relayed v1 transaction with contract call with MoveBalance, with signal error")
+    print("## Intra-shard, relayed v1 transaction with contract call with MoveBalance, with signal error")
     controller.send(controller.create_relayed_v1_with_contract_call_with_move_balance_with_signal_error(
         relayer=accounts.get_user(shard=0, index=0),
         sender=accounts.get_user(shard=0, index=1),
         contract=accounts.get_contract_address("adder", 0, 0),
         amount=1
     ))
 
+    print("## Intra-shard ClaimDeveloperRewards on directly owned contract")
+    controller.send(controller.create_claim_developer_rewards_on_directly_owned_contract(
+        sender=accounts.get_user(shard=0, index=0),
+        contract=accounts.get_contract_address("adder", 0, 0),
+    ))
+
+    print("## Cross-shard ClaimDeveloperRewards on directly owned contract")
+    controller.do_change_contract_owner(
+        contract=accounts.get_contract_address("adder", shard=1, index=0),
+        new_owner=accounts.get_user(shard=0, index=0)
+    )
+
+    controller.send(controller.create_claim_developer_rewards_on_directly_owned_contract(
+        sender=accounts.get_user(shard=0, index=0),
+        contract=accounts.get_contract_address("adder", shard=1, index=0),
+    ))
+
+    print("## ClaimDeveloperRewards on child contract (owned by a contract); user & parent contract in same shard")
+    controller.send(controller.create_claim_developer_rewards_on_child_contract(
+        sender=accounts.get_user(shard=0, index=0),
+        parent_contract=accounts.get_contract_address("developerRewards", shard=0, index=0),
+    ))
+
+    print("## ClaimDeveloperRewards on child contract (owned by a contract); user & parent contract in different shards")
+    controller.send(controller.create_claim_developer_rewards_on_child_contract(
+        sender=accounts.get_user(shard=0, index=0),
+        parent_contract=accounts.get_contract_address("developerRewards", shard=1, index=0),
+    ))
+
 
 class BunchOfAccounts:
     def __init__(self, configuration: Configuration, memento: "Memento") -> None:
         self.configuration = configuration
         self.mnemonic = Mnemonic(configuration.users_mnemonic)
         self.sponsor = self._create_sponsor()
         self.users: List[Account] = []
-        self.users_by_shard: List[List[Account]] = [[], [], []]
         self.users_by_bech32: Dict[str, Account] = {}
         self.contracts: List[SmartContract] = []
-        self.contracts_by_shard: List[List[SmartContract]] = [[], [], []]
-
-        address_computer = AddressComputer()
 
-        for i in range(32):
+        for i in range(128):
```

Comment:
`128` seems like a magic number, could be extracted as a constant in the next PR.